specifyweb/backend/setup_tool/setup_tasks.py

```diff
@@ -0,0 +1,171 @@
+"""
+A Celery task for setting up the database in the background.
+"""
+
+from specifyweb.celery_tasks import app
+from typing import Tuple, Optional
+from celery.result import AsyncResult
+from specifyweb.backend.setup_tool import api
+from django.db import transaction
+import threading
+from specifyweb.specify.models_utils.model_extras import PALEO_DISCIPLINES, GEOLOGY_DISCIPLINES
+import traceback
+
+from uuid import uuid4
+import logging
+logger = logging.getLogger(__name__)
+
+# Keep track of the currently running setup task. There should only ever be one.
+_active_setup_task_id: Optional[str] = None
+_active_setup_lock = threading.Lock()
+
+# Keep track of last error.
+_last_error: Optional[str] = None
+_last_error_lock = threading.Lock()
+
+class MissingWorkerError(Exception):
+    """Raised when worker is not running."""
+    pass
+
+def setup_database_background(data: dict) -> str:
+    global _active_setup_task_id, _last_error
+
+    # Clear any previous error logs.
+    set_last_setup_error(None)
+
+    if not is_worker_alive():
+        set_last_setup_error("The Specify Worker is not running.")
+        raise MissingWorkerError("The Specify Worker is not running.")
+
+    task_id = str(uuid4())
+    logger.debug(f'task_id: {task_id}')
+
+    args = [data]
+
+    task = setup_database_task.apply_async(args, task_id=task_id)
+
+    with _active_setup_lock:
+        _active_setup_task_id = task.id
+    
+    return task.id
+
+def is_worker_alive():
+    """Pings the worker to see if its running."""
+    try:
+        res = app.control.inspect(timeout=1).ping()
+        return bool(res)
+    except Exception:
+        return False
```

Comment:
(optional)

Can this be moved to a dedicated worker-related file? (Even if its just in the main [celery_tasks.py](https://github.com/specify/specify7/blob/main/specifyweb/celery_tasks.py)): I'd love to keep this visible and in a central location for other components to easily find and use this. 

A general utility like this would be perfect for something like https://github.com/specify/specify7/issues/6766